File: scala/src/main/scala/org/openapitools/models/SearchGroceryProductsByUPC200Response.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import org.openapitools.models.BigDecimal
import org.openapitools.models.SearchGroceryProductsByUPC200ResponseIngredientsInner
import org.openapitools.models.SearchGroceryProductsByUPC200ResponseNutrition
import org.openapitools.models.SearchGroceryProductsByUPC200ResponseServings
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param title 
 * @param badges 
 * @param importantBadges 
 * @param breadcrumbs 
 * @param generatedText 
 * @param imageType 
 * @param ingredientCount 
 * @param ingredientList 
 * @param ingredients 
 * @param likes 
 * @param nutrition 
 * @param price 
 * @param servings 
 * @param spoonacularScore 
 */
case class SearchGroceryProductsByUPC200Response(id: Int,
                title: String,
                badges: Seq[String],
                importantBadges: Seq[String],
                breadcrumbs: Seq[String],
                generatedText: String,
                imageType: String,
                ingredientCount: Option[Int],
                ingredientList: String,
                ingredients: Set[SearchGroceryProductsByUPC200ResponseIngredientsInner],
                likes: BigDecimal,
                nutrition: SearchGroceryProductsByUPC200ResponseNutrition,
                price: BigDecimal,
                servings: SearchGroceryProductsByUPC200ResponseServings,
                spoonacularScore: BigDecimal
                )

object SearchGroceryProductsByUPC200Response {
    /**
     * Creates the codec for converting SearchGroceryProductsByUPC200Response from and to JSON.
     */
    implicit val decoder: Decoder[SearchGroceryProductsByUPC200Response] = deriveDecoder
    implicit val encoder: ObjectEncoder[SearchGroceryProductsByUPC200Response] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/search_grocery_products_by_upc_200_response.clj
(ns spoonacular-api.specs.search-grocery-products-by-upc-200-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.ingredient-basics :refer :all]
            [spoonacular-api.specs.search-grocery-products-by-upc-200-response-nutrition :refer :all]
            [spoonacular-api.specs.search-grocery-products-by-upc-200-response-servings :refer :all]
            )
  (:import (java.io File)))


(def search-grocery-products-by-upc-200-response-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :badges) (s/coll-of string?)
   (ds/req :importantBadges) (s/coll-of string?)
   (ds/req :breadcrumbs) (s/coll-of string?)
   (ds/req :generatedText) string?
   (ds/req :imageType) string?
   (ds/opt :ingredientCount) int?
   (ds/req :ingredientList) string?
   (ds/req :ingredients) (s/coll-of ingredient-basics-spec)
   (ds/req :likes) float?
   (ds/req :nutrition) search-grocery-products-by-upc-200-response-nutrition-spec
   (ds/req :price) float?
   (ds/req :servings) search-grocery-products-by-upc-200-response-servings-spec
   (ds/req :spoonacularScore) float?
   })

(def search-grocery-products-by-upc-200-response-spec
  (ds/spec
    {:name ::search-grocery-products-by-upc-200-response
     :spec search-grocery-products-by-upc-200-response-data}))
